packages/launch/src/hooks/use-domain-selection.ts

```diff
@@ -1,15 +1,64 @@
 /**
  * External dependencies
  */
+import * as React from 'react';
 import { useDispatch, useSelect } from '@wordpress/data';
 import type { DomainSuggestions } from '@automattic/data-stores';
 
 /**
  * Internal dependencies
  */
-import { LAUNCH_STORE } from '../stores';
+import { LAUNCH_STORE, SITE_STORE, DOMAIN_SUGGESTIONS_STORE } from '../stores';
+import LaunchContext from '../context';
+import { isDomainProduct } from '../utils';
+import type { Product, DomainProduct } from '../utils';
 
-export function useDomainSelection() {
+export function useDomainProductFromCart(): DomainProduct | undefined {
+	const { siteId } = React.useContext( LaunchContext );
+	const { getCart } = useDispatch( SITE_STORE );
+
+	const [ domainProductFromCart, setDomainProductFromCart ] = React.useState<
+		DomainProduct | undefined
+	>( undefined );
+
+	React.useEffect( () => {
+		( async function () {
+			const cart = await getCart( siteId );
+			const domainProduct = cart.products?.find( ( item: Product ) => isDomainProduct( item ) );
+			setDomainProductFromCart( domainProduct );
+		} )();
+	}, [ siteId, getCart, setDomainProductFromCart ] );
+
+	return domainProductFromCart;
+}
```

Comment:
Suggested a wp-data friendly approach in https://github.com/Automattic/wp-calypso/pull/47999/files#diff-e7e38dec68b1a4011df8cce1217ea7c0554561fca279540153d2216f961b97f5R46-R57

As I mentioned also on the other related PR, this is not a blocker https://github.com/Automattic/wp-calypso/pull/47990#pullrequestreview-543965349

Reply:
Okay using resolvers! Gotcha. Keeping tabs on this.